diff -Naur a/python/samba/netcmd/testparm.py b/python/samba/netcmd/testparm.py
--- a/python/samba/netcmd/testparm.py	2023-01-09 18:46:35.000000000 +0700
+++ b/python/samba/netcmd/testparm.py	2023-01-09 22:55:10.696393600 +0700
@@ -145,14 +145,12 @@
         lockdir = lp.get("lockdir")
 
         if not os.path.isdir(lockdir):
-            logger.error("lock directory %s does not exist", lockdir)
-            valid = False
+            logger.warning("lock directory %s does not exist", lockdir)
 
         piddir = lp.get("pid directory")
 
         if not os.path.isdir(piddir):
-            logger.error("pid directory %s does not exist", piddir)
-            valid = False
+            logger.warning("pid directory %s does not exist", piddir)
 
         winbind_separator = lp.get("winbind separator")
 
diff -Naur a/source3/utils/testparm.c b/source3/utils/testparm.c
--- a/source3/utils/testparm.c	2023-01-09 18:46:35.000000000 +0700
+++ b/source3/utils/testparm.c	2023-01-09 22:56:05.840410700 +0700
@@ -326,9 +326,8 @@
 	}
 
 	if (!directory_exist_stat(lp_lock_directory(), &st)) {
-		fprintf(stderr, "ERROR: lock directory %s does not exist\n\n",
+		fprintf(stderr, "WARNING: lock directory %s does not exist\n\n",
 		       lp_lock_directory());
-		ret = 1;
 	} else if ((st.st_ex_mode & 0777) != 0755) {
 		fprintf(stderr, "WARNING: lock directory %s should have "
 				"permissions 0755 for browsing to work\n\n",
@@ -356,9 +355,8 @@
 	}
 
 	if (!directory_exist_stat(lp_pid_directory(), &st)) {
-		fprintf(stderr, "ERROR: pid directory %s does not exist\n\n",
+		fprintf(stderr, "WARNING: pid directory %s does not exist\n\n",
 		       lp_pid_directory());
-		ret = 1;
 	}
 
 	if (lp_passdb_expand_explicit()) {
