diff -Naur a/lib/ldb/wscript b/lib/ldb/wscript
--- a/lib/ldb/wscript
+++ b/lib/ldb/wscript
@@ -362,11 +362,6 @@
                      deps='cmocka ldb',
                      install=False)
 
-    bld.SAMBA_BINARY('test_ldb_comparison_fold',
-                     source='tests/test_ldb_comparison_fold.c',
-                     deps='cmocka ldb ldbwrap',
-                     install=False)
-
     bld.SAMBA_BINARY('ldb_key_value_test',
                      source='tests/ldb_key_value_test.c',
                      deps='cmocka ldb ldb_tdb_err_map',
